### ISSUE TITLE
Feature request: buffer parameter for bitarray

### ISSUE THREAD

**Issue by 0xf0f** [ilanschnell/bitarray#141]
Similar to the buffer parameter for numpy.ndarray, the ability to create bitarray backed by the memory of any object which implements the buffer protocol. An example use case for this would be to create bitarrays using memory mapped files:

```py
import bitarray
import mmap

with open('large_binary_file', 'r+b') as file:
    with mmap.mmap(file.fileno(), 0) as mapping:  # map the entire file
        array = bitarray.bitarray(buffer=mapping)  # create bitarray backed by the mapping
        array[::2] = True  # modifications to the bitarray are applied to the mapping
    # changes to the mapping will automatically be flushed back out to the file
```

**Comment by ilanschnell** [comment]
Thanks for the feature request.  It sounds like a useful feature.  The bitarray object already supports the buffer protocol - exporting the bitarrays buffer to a `memoryview` object.  However, it does not support importing an object which supports the buffer protocol.  I'm not sure how difficult this would be to achieve, but I'll look at into it some more.

**Comment by ilanschnell** [comment]
Thanks for requesting the buffer parameter.  I learned a lot about the buffer protocol.  All is implemented, tested, documented (including a memory mapped file example) and released in version 2.3.0.

**Comment by 0xf0f** [comment]
Fantastic work, thank you
